pkg/operator/featuregates/featuregate_controller_test.go

```diff
@@ -522,10 +558,89 @@ func TestFeatureGateController_sync(t *testing.T) {
 						"Eggplant", // known
 						"Five",     // known
 						"FoieGras", // known
+						"Seven",    // version unset
 						"Six",      // known
 					).
 					toFeatureGate()
 				if !reflect.DeepEqual(actual, expected) {
+					t.Fatalf("expected: %+v, actual %+v", spew.Sdump(expected), spew.Sdump(actual))
+				}
+			},
+		},
+		{
+			name:        "filter-given-minimum-version-not-set",
+			cvoVersions: []string{"current-version", "prior-version"},
+			firstFeatureGate: featureGateBuilder().
+				withFeatureSet(configv1.Default).
+				statusEnabled("current-version", "One", "Two", "Seven").
+				statusDisabled("current-version", "Apple", "Banana").
+				toFeatureGate(),
+			fields: fields{
+				processVersion: "current-version",
+			},
+			changeVerifier: func(t *testing.T, actions []kubetesting.Action, versionRecorder status.VersionGetter) {
+				if versionRecorder.GetVersions()[FeatureVersionName] != "current-version" {
+					t.Errorf("bad version: %v", versionRecorder.GetVersions())
+				}
+				if len(actions) != 1 {
+					t.Fatalf("bad changes: %v", actions)
+				}
+				updateAction := actions[0].(kubetesting.UpdateAction)
+				actual := updateAction.GetObject().(*configv1.FeatureGate)
+				expected := featureGateBuilder().
+					withFeatureSet(configv1.Default).
+					statusEnabled("current-version", "Five", "Six").
+					statusDisabled("current-version",
+						"Apple",    // known
+						"Banana",   // known
+						"Eggplant", // specific
+						"FoieGras", // specific
+						"One",      // known
+						"Seven",    // not set
+						"Two",      // known
+					).
+					toFeatureGate()
+				if !reflect.DeepEqual(actual, expected) {
+					t.Log(spew.Sdump(expected))
+					t.Fatal(spew.Sdump(actual))
+				}
+			},
+		},
+		{
+			name:        "add-given-minimum-version-set",
+			cvoVersions: []string{"current-version", "prior-version"},
+			firstFeatureGate: featureGateBuilder().
+				withFeatureSet(configv1.Default).
+				statusEnabled("current-version", "One", "Two", "Seven").
+				statusDisabled("current-version", "Apple", "Banana").
+				toFeatureGate(),
+			fields: fields{
+				processVersion: "current-version",
+			},
+			minimumKubeletVersion: "1.30.0",
```

Comment:
I was expecting seven to be enabled in this case? The min here is 1.30 which matches the version in the gate itself?

Reply:
🤦‍♂️ I can't read today apparently, all good